### PR TITLE
[#5593] improvement(iceberg): Add final to Iceberg table event members, fix return codes for rename and exists operations

### PR DESCRIPTION
**Pull request by orenccl**
### What changes were proposed in this pull request?

1. Add final to Iceberg table-related events and their members.

2. Modify the return codes in `IcebergTableOperations` and `IcebergTableRenameOperations` to align with the specification, and update the relevant unit tests.

### Why are the changes needed?

Close: #5593

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Unit test

### CODE REVIEW

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/listener/api/event/IcebergUpdateTableEvent.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
 @DeveloperApi
 public class IcebergUpdateTableEvent extends IcebergTableEvent {
 
-  private UpdateTableRequest updateTableRequest;
+  private final UpdateTableRequest updateTableRequest;
   private LoadTableResponse loadTableResponse;
```
Comment (FANNG1) [inline iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/listener/api/event/IcebergUpdateTableEvent.java:33]: add `final` to `loadTableResponse` too?

**Comment by FANNG1** [comment]
@orenccl thanks for your PR, could you fix exists return code in name space operation too?

File: iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/listener/api/event/IcebergUpdateTableEvent.java (hunk 1 above)
Comment (orenccl) [reply iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/listener/api/event/IcebergUpdateTableEvent.java]: Sorry, I missed that.

**Comment by orenccl** [comment]
Hi @FANNG1,

I checked `IcebergNamespaceOperations` and Iceberg's `catalogHandlers`, but didn't find anything like `namespaceExists`. It seems there isn't an existing function for that.

So, no modifications are needed here?

**Comment by FANNG1** [comment]
> Hi @FANNG1,
> 
> I checked `IcebergNamespaceOperations` and Iceberg's `catalogHandlers`, but didn't find anything like `namespaceExists`. It seems there isn't an existing function for that.
> 
> So, no modifications are needed here?

OOH, seems we forget to implement this interface, this is another issue.

**Review by FANNG1 — APPROVED** [review]

**Comment by FANNG1** [comment]
@orenccl , merged to main, thanks for your work and insight.

**Pull request merged**